MetaU Capstone Project/View Controllers/SearchTableViewController.m

```diff
@@ -0,0 +1,156 @@
+//
+//  SearchTableViewController.m
+//  Pods
+//
+//  Created by Rodjina Pierre Louis on 8/11/22.
+//
+
+#import "SearchTableViewController.h"
+#import "UIImageView+AFNetworking.h"
+#import "SpotifyiOS/SpotifyiOS.h"
+#import "SpotifyManager.h"
+#import "TrackCell.h"
+
+@interface SearchTableViewController ()
+
+@property (strong, nonatomic) NSDictionary *searchResults;
+
+@end
+
+@implementation SearchTableViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.tableView.delegate = self;
+    self.tableView.dataSource = self;
+    self.searchBar.delegate = self;
+    
+    [self loadSearchResults];
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+}
+
+- (void)loadSearchResults {
+    [[SpotifyManager shared] searchTrack:self.searchBar.text type:@"track" result:^(NSDictionary * _Nonnull dataRecieved) {
+        self.searchResults = [dataRecieved valueForKey:@"tracks"];
+    }];
+}
+
+- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText {
+    [NSObject cancelPreviousPerformRequestsWithTarget:self];
+
+    [self performSelector:@selector(loadSearchResults) withObject:searchText afterDelay:0.5];
+    [self.tableView reloadData];
+}
+
+// Number of rows
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return self.searchResults.count;
+}
+
+// Cells and Cell customization
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    
+    TrackCell *cell = [tableView dequeueReusableCellWithIdentifier:@"TrackCell" forIndexPath:indexPath];
+    
+    if (cell == nil) {
+        cell = [[TrackCell alloc] initWithStyle:UITableViewCellStyleSubtitle reuseIdentifier:@"TrackCell"];
+    }
```

Comment:
I think you want to avoid this.
`dequeueReusableCellWithIdentifier` will usually always return a non-null cell if you pass in the correct identifier.
For that, I would suggest that you declare the identifier as a constant like:

`static NSString *const kTrackCellIdentifier = @"TrackCell";`

Then use it like: 
`TrackCell *cell = [tableView dequeueReusableCellWithIdentifier:kTrackCellIdentifier forIndexPath:indexPath];`